src/index.ts

```diff
@@ -51,6 +52,10 @@ export class ReadmeBox {
       newContents
     })
 
+    if (opts.emptyCommits === false && content === replaced) {
+      return
+    }
```

Comment:
👍  Thanks for putting the `opts.emptyCommits` check first, potentially saves some string comparison time.